sky/cli.py

```diff
@@ -3876,6 +3950,459 @@ def spot_dashboard(port: Optional[int]):
+                       '--purge to clean up the controller.'
+                       f'{colorama.Style.RESET_ALL}'
+                       f'\nReason: {common_utils.format_exception(e)}.')
+        except (exceptions.NotSupportedError,
+                exceptions.ClusterOwnerIdentityMismatchError) as e:
+            message = str(e)
+        else:
+            message = (f'{colorama.Fore.GREEN}Teardown service {name}...done.'
+                       f'{colorama.Style.RESET_ALL}')
+            success_progress = True
+
+        progress.stop()
+        click.echo(message)
+        if success_progress:
+            progress.update(task, advance=1)
+        progress.start()
+
+    with progress:
+        subprocess_utils.run_in_parallel(_down_service, service_names)
+        progress.live.transient = False
+        # Make sure the progress bar not mess up the terminal.
+        progress.refresh()
+
+
+@serve.command('logs', cls=_DocumentedCodeCommand)
+@click.option(
+    '--follow/--no-follow',
+    is_flag=True,
+    default=True,
+    help=('Follow the logs of the job. [default: --follow] '
+          'If --no-follow is specified, print the log so far and exit.'))
+@click.option('--controller',
+              is_flag=True,
+              default=False,
+              required=False,
+              help='Show the controller logs of this service.')
+@click.option('--load-balancer',
+              is_flag=True,
+              default=False,
+              required=False,
+              help='Show the load balancer logs of this service.')
+@click.argument('service_name',
+                required=True,
+                type=str,
+                **_get_shell_complete_args(_complete_service_name))
+@click.argument('replica_id', required=False, type=int)
+@usage_lib.entrypoint
+def serve_logs(
+    service_name: str,
+    follow: bool,
+    controller: bool,
+    load_balancer: bool,
+    replica_id: Optional[int],
+):
+    """Tail the log of a service.
+
+    Example:
+
+    .. code-block:: bash
+
+        # Tail the controller logs of a service
+        sky serve logs --controller [SERVICE_ID]
+        \b
+        # Print the load balancer logs so far and exit
+        sky serve logs --load-balancer --no-follow [SERVICE_ID]
+        \b
+        # Tail the logs of replica 1
+        sky serve logs [SERVICE_ID] 1
+    """
+    have_replica_id = replica_id is not None
+    if (controller + load_balancer + have_replica_id) != 1:
+        raise click.UsageError(
+            'One and only one of --controller, --load-balancer, '
+            '[REPLICA_ID] can be specified.')
+    service_record = global_user_state.get_service_from_name(service_name)
+    if service_record is None:
+        click.secho(f'Service {service_name!r} not found.', fg='red')
+        return
+    controller_cluster_name = service_record['handle'].controller_cluster_name
+    if controller:
+        core.tail_logs(controller_cluster_name, job_id=1, follow=follow)
```

Comment:
Hard coding the `job_id` is not safe, and in the future it may not work if we add some retries.
Can we instead use the job_name?

Reply:
Now deprecated since we no longer use `sky logs` to tail controller / load balancer logs